modules/iam-read-only-policy/main.tf

```diff
@@ -0,0 +1,90 @@
+resource "aws_iam_policy" "policy" {
+  count = var.create_policy ? 1 : 0
+
+  name        = var.name
+  path        = var.path
+  description = var.description
+
+  policy = data.aws_iam_policy_document.combined.json
+
+  tags = var.tags
+}
+
+locals {
+  console_services = ["resource-groups", "tag", "health"]
+}
+
+data "aws_iam_policy_document" "allowed_services" {
+
```

Comment:
No resources or data sources should be created if `create_policy = false` is set. Add `count` to everything.

Reply:
intention of create_policy is to skip creation of iam policy and only generate json doc thus you need this data in both cases and that is why there is no count

Reply:
if I misunderstood your comment please give me an example of what you mean

Reply:
Normally, in other terraform-aws-modules, nothing is being created if `create_* = false` is defined. This way is compatible with situations when users want to control the creation of `all` or `nothing` (e.g. still handy with terragrunt).

I see what you mean in this example. Let's leave it as you have done already. I think it makes sense here.